src/lib/journey/stages/one-time-password.svelte

```diff
@@ -91,7 +116,11 @@
     />
   {/each}
 
-  {#if metadata?.step?.derived.isUserInputOptional || !metadata?.step?.derived.isStepSelfSubmittable()}
+  {#if buttons?.length}
+    <Button busy={journey?.loading} style="primary" type="submit" width="full">
+      <T key={buttons[0].text} />
```

Comment:
Using the presence of the `ConfirmationCallback` to inform rendering here. If it's not present, we fall back to the normal button logic.